### ISSUE TITLE
Thief-Detection : Person Detection is Thief or Not (Based on taking up an unusual path)

### ISSUE THREAD

**Issue by prathimacode-hub** [neha07kumari/Thief_Detection#6]
### Description

Welcome to 'Thief -Detection', good to see you here.

With this issue, the readers will get to know on how to detect whether a person is thief or not.

To get assigned to this issue, add your participant id mentioned in this [spreadsheet](https://docs.google.com/spreadsheets/d/1SX9FubA-n1ag_0n4WmzMXd6rq5lrkFmaAWl8fhHvTWM/edit#gid=0) and the approach you would follow.

Points to Note :

- The issues will be assigned on a first come first serve basis, 1 Issue == 1 PR.

- "Issue Title" and "PR Title" should be the same. Include issue number in PR template along with it.

- You should add up in respective folder according to the demands of the issue.

- Inclusion of Screenshot is must to approve 

This issue is only for 'SWOC' contributors

### Domain

Data Science

### Code of Conduct

- [X] I  will follow [Contributing Guidelines](https://github.com/neha07kumari/Thief_Detection/blob/main/CONTRIBUTING.md) & [Code of conduct](https://github.com/neha07kumari/Thief_Detection/blob/main/CODE_OF_CONDUCT.md) of this project.

**Comment by kkg2001** [comment]
I want to work on this issue
#6

**Comment by prathimacode-hub** [comment]
You can work on only 1 issue at a time so let me know of what you would choose first? @kkg2001

**Comment by kkg2001** [comment]
I want to work on issue#6

**Comment by prathimacode-hub** [comment]
Ok, provide the details mentioned to get it assigned. @kkg2001

**Comment by kkg2001** [comment]
Name-Krishna Kant Gupta
Participant ID:499
Discord-krish1234#9508
Project:Person Detection is thief or not
github id:@kkg2001

**Comment by prathimacode-hub** [comment]
Approach please. @kkg2001

**Comment by kkg2001** [comment]
Approach: First of all we will check the stranger's body movements and take into consideration his body language from the usual parameters. Secondly using opencv or any other technology relevant to this project we will check whether the person is nervous/curious or not keeping into consideration how normal people's face reactions. Time parameter like whether the stranger looks in a hurry or he is pretending to be in normal state and taking more time than usual so that nobody catches him. There are chances that he may not use the usual path to enter the house in that case the siren alarm should ring immediately and should confirm that he is certainly a thief. Tracing his finger prints or detecting his face could make work easier whether his history is there as a thief.

**Comment by prathimacode-hub** [comment]
This would be a long thing for this issue. Nice ideas you have showcased. But we already created other issues for considerations mentioned. For this issue, we shall do it with basic one. If person is not entering through a usual path, we detect him as thief and alarm would ring immediately. @kkg2001

**Comment by kkg2001** [comment]
Thanks. Ok I will work on only the basic one when the thief is entering through an unusual path

**Comment by prathimacode-hub** [comment]
Yes, once you finish this. You can take up an another issue related to it. Issue is assigned to @kkg2001

**Comment by prathimacode-hub** [comment]
Yours is a basic and will be interlinked issue for many other issues. @kkg2001

**Comment by kkg2001** [comment]
Ok I will make sure I put up my best efforts for this issue and then I will move over to other issues (probably #7)

**Comment by prathimacode-hub** [comment]
Great, you can proceed. All the best. @kkg2001

**Comment by kkg2001** [comment]
I am presuming that cctv cameras are installed on that path from where stranger will try to enter the house from an unusual path. I will share the code shortly and let me know the changes in it.
Thanks

**Comment by prathimacode-hub** [comment]
Ok sure. Yeah, with cctv. Without cctv, you have any idea to add? I don't think owners would keep cctv on unsual path. For now, this works @kkg2001

**Comment by kkg2001** [comment]
If by some mechanism we can measure the randomness of motion of the stranger coming by unusual path then its possible(something like a sensor or if we can predict its motion history). For example coming via window pane or jumping off the fence will have different motions.

**Comment by prathimacode-hub** [comment]
Yes, for this issue we shall proceed with this thing we had thought of. We can add this up later in issues for better proximity. @kkg2001

**Comment by kkg2001** [comment]
You mean we can proceed with this cctv thing what we actually presumed or something else?

**Comment by prathimacode-hub** [comment]
It's the same what we had thought earlier. Proceeding with CCTV thing. @kkg2001

**Comment by prathimacode-hub** [comment]
Yes, the same. I thought you would have started working. @kkg2001

**Comment by kkg2001** [comment]
I started back then but in the back of my mind was thinking about what if the cctv was not in the unusual path. Regarding the coding part I am facing some errors so I am fixing those.

**Comment by prathimacode-hub** [comment]
Ok for now we will get this done and later add enhancements to it. What are the errors you're getting? @kkg2001

**Comment by kkg2001** [comment]
'error: unrecognsied elements' , 'an exception has occured , use %tb to see the full traceback'

**Comment by kkg2001** [comment]
*arguments not elements

**Comment by prathimacode-hub** [comment]
The elements you have mentioned in arguments aren't recognizing. Have you defined them? @kkg2001

**Comment by kkg2001** [comment]
Yes I have defined them. I will still again look onto it. Also I think the packages are not installed properly.

**Comment by prathimacode-hub** [comment]
Ok, Understood libraries should be loaded for functionality to work. @kkg2001

**Comment by prathimacode-hub** [comment]
Kindly update the progress. Its been more than 10days. It should be done by now, if you're stuck anywhere we are here to help out. @kkg2001

**Comment by kkg2001** [comment]
Ok sure. Give me some time.

**Comment by prathimacode-hub** [comment]
Because most of the work initiates from this issue. @kkg2001

**Comment by kkg2001** [comment]
I am facing some issues. I tried running the code on jupyter notebook and
imported open cv and numpy at the starting and also initiated video capture
for video file but it showed neither any response nor any error and video
file is in .avi format.

**Comment by prathimacode-hub** [comment]
Oh, if you could share me out the link @kkg2001

**Comment by kkg2001** [comment]
import cv2
import numpy as np

body_classifier = cv2.CascadeClassifier(cv2.data.haarcascades+'haarcascade_fullbody.xml')


cap = cv2.VideoCapture('Thief.avi')


while cap.isOpened():
    
    
    ret, frame = cap.read()
    

    gray = cv2.cvtColor(frame, cv2.COLOR_BGR2GRAY)
    
    bodies = body_classifier.detectMultiScale(gray, 1.2, 3)
    
    
    for (x,y,w,h) in bodies:
        cv2.rectangle(frame, (x, y), (x+w, y+h), (0, 255, 255), 2)
        cv2.imshow('Thief', frame)

    if cv2.waitKey(): 
        break

cap.release()
cv2.destroyAllWindows()

**Comment by kkg2001** [comment]
This is the code which is giving no response

**Comment by kkg2001** [comment]
I made the required changes in the code and its working fine. Can I proceed towards merging the PR?

**Comment by prathimacode-hub** [comment]
oops, sorry I was unwell. I couldn't check out your messages. Hope error is solved? Yes, if it's working you can proceed to do so
 @kkg2001

**Comment by kkg2001** [comment]
I am stuck in this. What to do?
![2022-01-14 (1)](https://user-images.githubusercontent.com/82452511/149547475-7efb4490-34fa-4434-b6a3-d94489940ddd.png)

**Comment by prathimacode-hub** [comment]
I guess your forked repo and main repo is not even. That's why it isn't dedicating the changes made.  Make it even and then push it back @kkg2001

**Comment by kkg2001** [comment]
First I have to fork and clone the repo in my system. I created the new
branch and upstream link was done too. After that should I add my folder
which contains code for this issue to the cloned repo in my system?

**Comment by prathimacode-hub** [comment]
Yes @kkg2001
